src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,136 @@
+                    Todo newTodo = new Todo(newtodo);
+                    tasks.add(newTodo);
+                    System.out.println("-------------------------------------------------------------------------");
+                    System.out.println("Got it! I've added this task: " + newTodo.getDescription());
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                } catch(IndexOutOfBoundsException e){
+                    ex.missingDescription("todo");
+                } finally{
+                    System.out.println("What else should I do for you?");
+                }
+            }
+            else if(userInput.startsWith("deadline")){
+                try{
+                    String[] input = userInput.split("/by");
+                    String ddate = input[1].trim();
+                    String deadline = input[0].split("deadline")[1].trim();
+                    Deadline newDeadline = new Deadline(deadline, ddate);
+                    tasks.add(newDeadline);
+                    System.out.println("-------------------------------------------------------------------------");
+                    System.out.println("Got it! I've added this task: ");
+                    System.out.println(newDeadline.getDescription());
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                } catch(IndexOutOfBoundsException e){
+                    ex.missingDescription("deadline");
+                } finally{
+                    System.out.println("What else should I do for you?");
+                }
+            }
+            else if(userInput.startsWith("event")){
+                try{
+                    String[] input = userInput.split("/at");
+                    String eventDate = input[1].trim();
+                    String event = input[0].split("event")[1].trim();
+                    Event newEvent = new Event(event, eventDate);
+                    tasks.add(newEvent);
+                    System.out.println("-------------------------------------------------------------------------");
+                    System.out.println("Got it! I've added this task: ");
+                    System.out.println(newEvent.getDescription());
+                    System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+                } catch(IndexOutOfBoundsException e){
+                    ex.missingDescription("event");
+                } finally{
+                    System.out.println("What else should I do for you?");
+                }
+            }
+            else{
+                Task newTask = new Task(userInput);
+                tasks.add(newTask);
+                System.out.println("added: "+ userInput);
+            }
+        }while(!userInput.equals("bye"));
+
+        System.out.println("Bye. Hope to see you again soon!");
+    }
```

Comment:
Avoid long methods (> 30 LoCs). Consider refactoring and SLAP to improve code readability.